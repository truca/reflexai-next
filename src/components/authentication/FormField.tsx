interface FormFieldProps {
  label: string;
  placeholder: string;
  value: string;
  onChange: (e: React.SyntheticEvent) => void;
}

export default function FormField({
  label,
  placeholder,
  value,
  onChange,
}: FormFieldProps) {
  return (
    <div>
      <label htmlFor={label} className="block text-gray-700">
        {label}
      </label>
      <input
        value={value}
        onChange={onChange}
        type="text"
        id={label}
        placeholder={placeholder}
        className="w-full px-4 py-3 rounded-lg bg-gray-200 mt-2 border focus:border-blue-500 focus:bg-white focus:outline-none"
        autoFocus
        required
      />
    </div>
  );
}
