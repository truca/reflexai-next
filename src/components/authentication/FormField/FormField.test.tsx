import FormField from "./index";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";

describe("FormField", () => {
  it("renders correctly the empty state", () => {
    const tree = renderer
      .create(
        <FormField
          label="Field"
          placeholder="Fill the field"
          value=""
          onChange={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly a state with value", () => {
    const tree = renderer
      .create(
        <FormField
          label="Field"
          placeholder="Fill the field"
          value="value for the field"
          onChange={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onChange function", () => {
    const handleOnChange = jest.fn();
    render(
      <FormField
        label="Field"
        placeholder="Fill the field"
        value=""
        onChange={handleOnChange}
      />
    );
    const input = screen.getByLabelText(/Field/i);
    fireEvent.change(input, { target: { value: "23" } });
    expect(handleOnChange).toHaveBeenCalled();
  });
});
