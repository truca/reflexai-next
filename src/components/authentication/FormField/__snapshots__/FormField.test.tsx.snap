// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField it renders correctly 1`] = `
<div>
  <label
    className="block text-gray-700"
    htmlFor="Field"
  >
    Field
  </label>
  <input
    autoFocus={true}
    className="w-full px-4 py-3 rounded-lg bg-gray-200 mt-2 border focus:border-blue-500 focus:bg-white focus:outline-none"
    id="Field"
    onChange={[Function]}
    placeholder="Fill the field"
    required={true}
    type="text"
    value=""
  />
</div>
`;

exports[`FormField renders correctly a state with value 1`] = `
<div>
  <label
    className="block text-gray-700"
    htmlFor="Field"
  >
    Field
  </label>
  <input
    autoFocus={true}
    className="w-full px-4 py-3 rounded-lg bg-gray-200 mt-2 border focus:border-blue-500 focus:bg-white focus:outline-none"
    id="Field"
    onChange={[Function]}
    placeholder="Fill the field"
    required={true}
    type="text"
    value="value for the field"
  />
</div>
`;

exports[`FormField renders correctly a state with value 2`] = `
<div>
  <label
    className="block text-gray-700"
    htmlFor="Field"
  >
    Field
  </label>
  <input
    autoFocus={true}
    className="w-full px-4 py-3 rounded-lg bg-gray-200 mt-2 border focus:border-blue-500 focus:bg-white focus:outline-none"
    id="Field"
    onChange={[Function]}
    placeholder="Fill the field"
    required={true}
    type="text"
    value="value for the field"
  />
</div>
`;

exports[`FormField renders correctly the empty state 1`] = `
<div>
  <label
    className="block text-gray-700"
    htmlFor="Field"
  >
    Field
  </label>
  <input
    autoFocus={true}
    className="w-full px-4 py-3 rounded-lg bg-gray-200 mt-2 border focus:border-blue-500 focus:bg-white focus:outline-none"
    id="Field"
    onChange={[Function]}
    placeholder="Fill the field"
    required={true}
    type="text"
    value=""
  />
</div>
`;
